#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Incremental PID:
# KP_ROT: 0.15
# TI_ROT: 0.35
# TD_ROT: 0.015

gen.add("KP_ROT", double_t, 0, "Rotation KP", 0.15,  0,  1.0)
gen.add("TI_ROT", double_t, 0, "Rotation TI", 1.5,  0,  100.0)
gen.add("TD_ROT", double_t, 0, "Rotation TD", 0.0, 0,  1.0)

gen.add("P_ANG_DST", double_t, 0, "Angle to goal filter", 0.95, 0, 1.0)
gen.add("P_ANG_THT", double_t, 0, "Curent angle goal filter", 0.7, 0, 1.0)
gen.add("UI_LIMIT_ROT", double_t, 0, "Integral control limit", 1.0,  0, 2.0)

gen.add("KP_FWD", double_t, 0, "Moving forward KP", 1.0,  0,  1000.0)
gen.add("TI_FWD", double_t, 0, "Moving forward TI", 0.7,  0,  1000.0)
gen.add("TD_FWD", double_t, 0, "Moving forward TD", 0.0, 0,  100.0)

gen.add("P_FWD_DST", double_t, 0, "Moving forward destination filter", 0.8, 0, 1.0)
gen.add("P_FWD_CUR", double_t, 0, "Current distance filter", 0.7, 0, 1.0)
gen.add("UI_LIMIT_FWD", double_t, 0, "Integral control limit", 0.3,  0, 100.0)

exit(gen.generate(PACKAGE, "dynamic_tutorials", "DynRecPID"))
